daemon/logger/factory.go

```diff
@@ -51,18 +54,21 @@ func (lf *logdriverFactory) register(name string, c Creator) error {
 	return nil
 }
 
-func (lf *logdriverFactory) driverRegistered(name string) bool {
+func (lf *logdriverFactory) driverRegistered(name string) (bool, error) {
 	lf.m.Lock()
 	_, ok := lf.registry[name]
 	lf.m.Unlock()
 	if !ok {
 		if pluginGetter != nil { // this can be nil when the init functions are running
-			if l, _ := getPlugin(name, plugingetter.Lookup); l != nil {
-				return true
+			l, err := getPlugin(name, plugingetter.Lookup)
+			if err != nil {
+				return false, err
+			} else if l != nil {
```

Comment:
No need for the `else if`

```go
if err != nil {
    return false, err
}
if l != nil {
    return true, nil
```

Same for other sections

Reply:
@cpuguy83 I solved this code =)